pkg/clustermesh/mcsapi/types/mcsapiservicespec.go

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	mcsapiv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/kvstore"
+	"github.com/cilium/cilium/pkg/kvstore/store"
+)
+
+var (
+	// ServiceExportStorePrefix is the kvstore prefix of the shared store
+	//
+	// WARNING - STABLE API: Changing the structure or values of this will
+	// break backwards compatibility
+	ServiceExportStorePrefix = path.Join(kvstore.BaseKeyPrefix, "state", "serviceexports", "v1")
+)
+
+type MCSAPIServiceSpec struct {
+	// Cluster is the cluster name the service is configured in
+	Cluster string `json:"cluster"`
+
+	// Name is the name of the ServiceImport or ServiceExport parent resources.
+	Name string `json:"name"`
+
+	// Namespace is the cluster namespace the service is configured in
+	Namespace string `json:"namespace"`
+
+	// ExportCreationTimestamp is the timestamp representing when the
+	// ServiceExport object was created. It is used for conflict resolution.
+	ExportCreationTimestamp metav1.Time `json:"exportCreationTimestamp"`
+
+	// Ports are the list of ports of the Service in MCS API format
+	Ports []mcsapiv1alpha1.ServicePort `json:"ports"`
+
+	// Type defines the type of this service.
+	// Must be ClusterSetIP or Headless.
+	Type mcsapiv1alpha1.ServiceImportType `json:"type"`
+
+	// Supports "ClientIP" and "None". Used to maintain session affinity.
+	// Enable client IP based session affinity.
+	// Must be ClientIP or None.
+	// Defaults to None.
+	// Ignored when type is Headless
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	SessionAffinity corev1.ServiceAffinity `json:"sessionAffinity,omitempty"`
```

Comment:
Nit. Could it make sense to remove `omitempty` if we always expect it to be present?